src/main/java/dev/zarr/zarrjava/core/Array.java

```diff
@@ -10,27 +11,73 @@
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
+import java.io.IOException;
 import java.nio.ByteBuffer;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.Arrays;
 import java.util.stream.Stream;
 
-public interface Array {
+public abstract class Array extends AbstractNode {
 
-    ArrayMetadata metadata();
-    
-    StoreHandle storeHandle();
+    protected CodecPipeline codecPipeline;
+    protected abstract ArrayMetadata metadata();
```

Comment:
Could this be made public (same for Group.metadata()), then the public metadata field in the implementation could be made private/protected instead? If not it's tricky to access the metadata (need explicit cast to v2.Array or v3.Array and access public metadata field).